### ISSUE TITLE
Products can't be displayed anymore after searching with German language enabled

### ISSUE THREAD

**Issue by ronnil1** [magento/magento2#37379]
Products are no longer displayed in the admin backend when searching for products with German language enabled.

### Preconditions (*)

1. Magento (community-edition) ver. 2.4.5-p1
2. Ubuntu 18.04, Apache 2.4.29, PHP8.1, MySQL 5.7.41
3. Additional: German language, € as currency symbol and important: Decimal separator is a comma (",").

### Steps to reproduce (*)

Once you have installed a German language pack:

1. Navigate to your admin backend
2. Log in 
3. In the top right corner click on your admin user name and select "Account Setting (<your admin user name>)"
4. Change "Interface Locale" to "Deutsch (Deutschland) / German (Germany)", type in your current password and save the settings by clicking the orange button "Save Account" in the top right corner.
5. In the left panel select "Catalog" (Katalog) -> Inventory: Products (Inventar: Produkte)

You now should see a list of your products and on the left above the article list you should see a search bar ("Search by keyword" / "Suche nach Schlagwort").

6. Select the search bar and type in a keyword and press Enter on your keyboard.

### Expected result (*)

1. If that keyword would result in some found records you should see a list of products that fit your keyword.

### Actual result (*)

With active German language and therefore using a comma as decimal separator the site shows a Warning pop-up: "Achtung: Es ist ein Fehler aufgetreten" (Attention: An error occurred). Underneath that pop-up there is also a message with lighter red background color which states: "Something went wrong with processing the default view and we have restored the filter to its original state.".
The search bar and the article list is greyed out and you can't select anything. You also can't remove the search filters. If you reload the page the pop-up appears again and a loading spinner shows up and spins infinitely.
![no_products_after_search](https://user-images.githubusercontent.com/3594152/232157391-2d71799d-505a-4356-9f0a-4f427b709b18.png)

In var/log/system.log there is an exception backtrace "Exception: Warning: A non-numeric value encountered". Which means some function/method expected a period (or full stop) (e.g. 12.34(€)) as decimal separator but got a comma (e.g. 12,34(€)). The full backtrace is:

```
main.CRITICAL: Exception: Warning: A non-numeric value encountered in /.../htdocs/vendor/magento/module-bundle/Ui/DataProvider/Product/Modifier/SpecialPriceAttributes.php on line 80 in /.../htdocs/vendor/magento/framework/App/ErrorHandler.php:62
Stack trace:
#0 /.../htdocs/vendor/magento/module-bundle/Ui/DataProvider/Product/Modifier/SpecialPriceAttributes.php(80): Magento\Framework\App\ErrorHandler->handler(2, 'A non-numeric v...', '/var/www/share/...', 80)
#1 /.../htdocs/vendor/magento/module-catalog/Ui/DataProvider/Product/ProductDataProvider.php(93): Magento\Bundle\Ui\DataProvider\Product\Modifier\SpecialPriceAttributes->modifyData(Array)
#2 /.../htdocs/vendor/magento/module-ui/Component/Listing.php(38): Magento\Catalog\Ui\DataProvider\Product\ProductDataProvider->getData()
#3 /.../htdocs/vendor/magento/framework/View/Element/UiComponent/Context.php(250): Magento\Ui\Component\Listing->getDataSourceData()
#4 /.../htdocs/vendor/magento/framework/View/Element/UiComponent/ContentType/Json.php(69): Magento\Framework\View\Element\UiComponent\Context->getDataSourceData(Object(Magento\Ui\Component\Listing))
#5 /.../htdocs/vendor/magento/module-ui/Component/AbstractComponent.php(158): Magento\Framework\View\Element\UiComponent\ContentType\Json->render(Object(Magento\Ui\Component\Listing), 'templates/listi...')
#6 /.../htdocs/vendor/magento/module-ui/Controller/Adminhtml/Index/Render.php(91): Magento\Ui\Component\AbstractComponent->render()
#7 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Ui\Controller\Adminhtml\Index\Render->execute()
#8 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->___callParent('execute', Array)
#9 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->Magento\Framework\Interception\{closure}()
#10 /.../htdocs/generated/code/Magento/Ui/Controller/Adminhtml/Index/Render/Interceptor.php(23): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->___callPlugins('execute', Array, Array)
#11 /.../htdocs/vendor/magento/framework/App/Action/Action.php(111): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->execute()
#12 /.../htdocs/vendor/magento/module-backend/App/AbstractAction.php(151): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#13 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#14 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->___callParent('dispatch', Array)
#15 /.../htdocs/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(145): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#16 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#17 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#18 /.../htdocs/generated/code/Magento/Ui/Controller/Adminhtml/Index/Render/Interceptor.php(32): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->___callPlugins('dispatch', Array, Array)
#19 /.../htdocs/vendor/magento/framework/App/FrontController.php(245): Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#20 /.../htdocs/vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController->getActionResponse(Object(Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor), Object(Magento\Framework\App\Request\Http))
#21 /.../htdocs/vendor/magento/framework/App/FrontController.php(147): Magento\Framework\App\FrontController->processRequest(Object(Magento\Framework\App\Request\Http), Object(Magento\Ui\Controller\Adminhtml\Index\Render\Interceptor))
#22 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#23 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#24 [internal function]: Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#25 /.../htdocs/vendor/m2epro/magento2-extension/Plugin/AbstractPlugin.php(36): call_user_func_array(Object(Closure), Array)
#26 /.../htdocs/vendor/m2epro/magento2-extension/Plugin/HealthStatus/Magento/Framework/App/FrontController.php(45): Ess\M2ePro\Plugin\AbstractPlugin->execute('dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Array)
#27 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(135): Ess\M2ePro\Plugin\HealthStatus\Magento\Framework\App\FrontController->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#28 /.../htdocs/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#29 /.../htdocs/generated/code/Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, NULL)
#30 /.../htdocs/vendor/magento/framework/App/Http.php(116): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#31 /.../htdocs/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#32 /.../htdocs/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#33 {main} [] [] 
```

**Comment by ronnil1** [comment]
> Hi @ronnil1. Thank you for your report. To speed up processing of this issue, make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, Add a comment to the issue:
> [...]

It is not possible to reproduce it without a German language pack, or any other language packs where the decimal separator is a comma. In your standard instance there is only English and therefore the decimal separator is a period/full stop.

How should we deal with this issue here, since the German language and German price display is important for usability?

**Comment by engcom-Bravo** [comment]
Hi @ronnil1,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


**Steps to reproduce**

Once you have installed a German language pack:

- Navigate to your admin backend
- Log in
- In the top right corner click on your admin user name and select "Account Setting ()"
- Change "Interface Locale" to "Deutsch (Deutschland) / German (Germany)", type in your current password and save the settings by clicking the orange button "Save Account" in the top right corner.
- In the left panel select "Catalog" (Katalog) -> Inventory: Products (Inventar: Produkte)
 You now should see a list of your products and on the left above the article list you should see a search bar ("Search by keyword" / "Suche nach Schlagwort").

- Select the search bar and type in a keyword and press Enter on your keyboard.



<img width="1479" alt="Screenshot 2023-04-17 at 1 38 01 PM" src="https://user-images.githubusercontent.com/51680745/232475603-ff9b9377-5a7b-48e3-b054-b200b4eb0ce3.png">



<img width="1464" alt="Screenshot 2023-04-17 at 1 38 17 PM" src="https://user-images.githubusercontent.com/51680745/232475647-c061cfd4-7089-4333-8ad8-e185b7b70787.png">



Kindly recheck the behavior on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.

Thanks.

**Comment by ronnil1** [comment]
Hey @engcom-Bravo,

thank you for triaging and checking on this issue. It seems that this issue is related to bundled products (and maybe special prices) in Magento ver. 2.3.5-p1 and -p2

Especially in `vendor/magento/module-bundle/Ui/DataProvider/Product/Modifier/SpecialPriceAttributes.php`, respectively in `app/code/Magento/Bundle/Ui/DataProvider/Product/Modifier/SpecialPriceAttributes.php` on line 80.
Since you cannot reproduce this issue on Magento 2.4-develop I checked the log for this file:

```
git log 2.4.5-p1..2.4-develop -- app/code/Magento/Bundle/Ui/DataProvider/Product/Modifier/SpecialPriceAttributes.php
```

And that gave me only one commit:

```
commit 9456a2659f3466046e1d5c7eb45ae60bd4703934
Author: Buba Suma <soumah@adobe.com>
Date:   Wed Jul 20 17:05:56 2022 -0500

    ACP2E-1038: A non-numeric value encountered Error for Bundles with special price over 1000 {Premier Support}
```


Further checks revealed that the commit occurs for the first time (or is merged) in version 2.4.6, so I copied that `SpecialPriceAttributes.php` file from Magento ver. 2.4.6 and replaced the one from Magento 2.4.5-p1 with it.

**Result:**
That commit also solved this issue here and that's why you cannot reproduce it on 2.4-develop.

––––––––––––

So my question is: Is Magento 2.4.5 not supported anymore and one has to upgrade to 2.4.6, or will there be a another patch for 2.4.5?

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @ronnil1,

Thanks for your update.

As I can see this issue got fixed in the scope of the internal Jira ticket ACP2E-1038 by the internal team.

Based on the Jira ticket, the target version is 2.4.6.

Kindly upgrade to latest version of Magento i.e, Magento 2.4.6.


Thanks.
